usersession/agent/session_agent.go

```diff
@@ -291,9 +310,25 @@ Loop:
 	return nil
 }
 
+// handleNotifications blocks handling notifications.
```

Comment:
nitpick ^ 3

```suggestion
// handleNotifications handles notifications in a blocking manner.
```

it's a bit meh either way, but this way seems slightly clearer to me